### PR TITLE
[GPU] ScatterUpdate blocked layout support

### PR DESCRIPTION
**Pull request by okhovan**
- added blocked layouts support;
- added unit tests for blocked layouts.

### Tickets:
 - 85177

### CODE REVIEW

File: src/plugins/intel_gpu/src/graph/impls/ocl/scatter_update.cpp [hunk 1]
```diff
@@ -75,17 +75,31 @@ struct scatter_update_impl : typed_primitive_impl_ocl<scatter_update> {
...
+        format::bs_fs_zyx_bsv32_fsv16,
+        format::bfwzyx
+    };
+
+    std::set<std::tuple<data_types, format::type>> keys;
+    for (const auto& t : types) {
+        for (const auto& f : formats) {
+            keys.emplace(t, f);
+        }
+    }
```
Comment (tgubanova-lohika) [inline src/plugins/intel_gpu/src/graph/impls/ocl/scatter_update.cpp:101]: no need to make keys manually, we already have overloaded map::add method with two params types and layouts

File: src/plugins/intel_gpu/src/kernel_selector/core/actual_kernels/scatter_update/scatter_update_kernel_ref.cpp [hunk 2]
```diff
@@ -48,6 +48,28 @@ ParamsKey ScatterUpdateKernelRef::GetSupportedKey() const {
...
+    k.EnableOutputLayout(DataLayout::b_fs_zyx_fsv16);
+    k.EnableOutputLayout(DataLayout::bs_fs_zyx_bsv16_fsv16);
+    k.EnableInputLayout(DataLayout::b_fs_zyx_fsv32);
+    k.EnableOutputLayout(DataLayout::b_fs_zyx_fsv32);
+    k.EnableInputLayout(DataLayout::bs_fs_zyx_bsv16_fsv32);
+    k.EnableOutputLayout(DataLayout::bs_fs_zyx_bsv16_fsv32);
+    k.EnableInputLayout(DataLayout::bs_fs_zyx_bsv32_fsv32);
+    k.EnableOutputLayout(DataLayout::bs_fs_zyx_bsv32_fsv32);
+    k.EnableInputLayout(DataLayout::bs_fs_zyx_bsv32_fsv16);
+    k.EnableOutputLayout(DataLayout::bs_fs_zyx_bsv32_fsv16);
```
Comment (tgubanova-lohika) [inline src/plugins/intel_gpu/src/kernel_selector/core/actual_kernels/scatter_update/scatter_update_kernel_ref.cpp:72]: as we mostly use all the formats here I suppose there is not critical to use EnableAllInputLayouts, EnableAllOutputsLayouts

**Review by tgubanova-lohika — COMMENTED** [review]
could you add operation to program.cpp

**Comment by wilson-seok** [comment]
@okhovan Build error happened in build_rhel82_release. Please let me know if you need full log.
image

**Comment by okhovan** [comment]
> @okhovan Build error happened in build_rhel82_release. Please let me know if you need full log. image

Yes, it would be great if you provide the full log. It is really strange - implementation_map::add() overload that causes compiler error was merged quite a long time ago... Maybe full log will sched some light on this PR issue. 
Thank you!

**Comment by tgubanova-lohika** [comment]
> > @okhovan Build error happened in build_rhel82_release. Please let me know if you need full log. image
> 
> Yes, it would be great if you provide the full log. It is really strange - implementation_map::add() overload that causes compiler error was merged quite a long time ago... Maybe full log will sched some light on this PR issue. Thank you!

could it be connected with fix which is not merged yet?
https://github.com/openvinotoolkit/openvino/pull/12043

**Comment by wilson-seok** [comment]
No performance regression in int8 models.

**Review by wilson-seok — APPROVED** [review]

**Pull request merged**